app/src/redux/actions/file.js

```diff
@@ -97,6 +97,30 @@ export function doStartDownload(uri, outpoint) {
   };
 }
 
+export function doStopDownloadingFile(uri, fileInfo) {
+  return dispatch  => {
+    let params = { status: 'stop' };
+    if (fileInfo.sd_hash) {
+      params.sd_hash = fileInfo.sd_hash; 
+    }
+    if (fileInfo.stream_hash) {
+      params.stream_hash = fileInfo.stream_hash;
+    }
+
+    Lbry.file_set_status(params).then(() => {
+      dispatch({
+        type: ACTIONS.DOWNLOADING_CANCELED,
+        data: {}
+      });
+    });
+    
+    NativeModules.LbryDownloadManager.stopDownload(uri, fileInfo.file_name);
+    
+    // Should also delete the file after the user stops downloading
+    dispatch(doDeleteFile(fileInfo.outpoint, uri));
```

Comment:
Can we keep the file half downloaded? I guess I'm not sure if this it would be valuable, but I might want to pause a large download and come back to it (if I leave wifi)

Reply:
Yes, I believe we can. In this case, I assume it would make sense to add a pause download button specifically for this.